core/aws_ddk_core/pipelines/pipeline.py

```diff
@@ -131,3 +134,28 @@ def add_rule(
             )
         )
         return self
+
+    def add_notifications(self, notifications_topic: Optional[ITopic] = None) -> "DataPipeline":
```

Comment:
one concern for me is that this method must be called before any stages that the user wants to consume notifications from